Jenkinsfile

```diff
@@ -38,41 +38,48 @@ pipeline {
     options {
         buildDiscarder(logRotator(daysToKeepStr: daysToKeep, numToKeepStr: numToKeep))
         timestamps()
-        timeout(time: 45, unit: 'MINUTES')
+    }
+
+    parameters {
+        booleanParam(name: 'RUN_QA', defaultValue: false, description: "Run QA Stage")
+        string(name: 'GENERATION_TEMPLATE', defaultValue: 'Zeebe 0.x.0', description: "Generation template for QA tests (the QA test will be run with this Zeebe version and Operate/Elasticsearch version from the generation template)")
     }
 
     stages {
         stage('Prepare') {
             steps {
-                script {
-                    commit_summary = sh([returnStdout: true, script: 'git show -s --format=%s']).trim()
-                    displayNameFull = "#" + BUILD_NUMBER + ': ' + commit_summary
-
-                    if (displayNameFull.length() <= 45) {
-                        currentBuild.displayName = displayNameFull
-                    } else {
-                        displayStringHardTruncate = displayNameFull.take(45)
-                        currentBuild.displayName = displayStringHardTruncate.take(displayStringHardTruncate.lastIndexOf(" "))
+                timeout(1) {
```

Comment:
What is 1 ? Seconds, Minutes hours?!

Reply:
So all units are in minutes. I can make that more verbose.
The times were chosen by looking at how much the phases take today. 

The main reason for the timeout is to deal with disconnects. Like if a node disconnects, usually the timeout kills the build and then the post build analysis discovers it was a disconnect. That is why a the timeouts are rather tight. If a phase usually takes 20 seconds, I don't want to wait for 15 minutes for the build to fail. 

But I can also set a minimum of 15 minutes for each step, sure.

Let me know what you prefer.

Reply:
>  I don't want to wait for 15 minutes for the build to fail.
Wouldn't the test fail immediately if a node dies?

Reply:
Sometimes yes, many times not. I dpn't fully understand why. It seems like the parent node is waiting for a callback from the agent it started, which never arrives.

Reply:
Yeah lets update the times to a bit higher value. Maybe 10 to meet in the middle :sweat_smile: The issue with such small timeouts is that you also easily scratch them if you just add some new tests.